crypto/x509/x509_vfy.c

```diff
@@ -3701,6 +3730,17 @@ static int build_chain(X509_STORE_CTX *ctx)
 
         num = sk_X509_num(ctx->chain);
         ctx->error_depth = num - 1;
+        /*
+         * If the top certificate is in trusted and the flag
+         * X509_V_FLAG_PARTIAL_CHAIN is set, we don't need to build
+         * the chain further
+         */
+        if (trust == X509_TRUST_UNTRUSTED && DANETLS_HAS_DANE_TA(dane))
+            trust = check_dane_pkeys(ctx);
+        if (trust == X509_TRUST_UNTRUSTED && num == ctx->num_untrusted)
+            trust = check_trust(ctx, num);
+        if (trust != X509_TRUST_UNTRUSTED)
+            break;
```

Comment:
How does this relate with the change? Is this needed and/or is it useful on its own?